client/my-sites/stats/stats-notices/commercial-site-upgrade-notice.tsx

```diff
@@ -77,8 +79,8 @@ const CommercialSiteUpgradeNotice = ( { siteId, isOdysseyStats }: StatsNoticePro
 			<NoticeBanner
 				level="info"
 				title={ translate( 'Upgrade to Stats Commercial' ) }
-				onClose={ dismissNotice }
-				hideCloseButton
+				onClose={ hideCloseButton ? () => {} : dismissNotice }
```

Comment:
I just realised that this component doesn't take in any props from outside so we could simplify this based on your last change:
```
onClose={ () => {} }
hideCloseButton
```

Reply:
If we want to keep the code to reuse in the future, could we just leave the prop `onClose` as `dismissNotice` since it [will not be executed](https://github.com/Automattic/wp-calypso/blob/5580d61a5a9ac33e396384682deb302992104206/packages/components/src/notice-banner/index.tsx#L83) when `hideCloseButton` is `true`?

Reply:
I don't think that either change impacts reusability. This is a component for a specific type of notice, not a shared notice component, so it can define the behaviour for all places where commercial notice is used.

I don't have strong opinions about either scenario, but it feels safer to avoid sending the click handler if we don't want the action to be executed, even if it's prevented in a child component.